app/controllers/groups_controller.rb

```diff
@@ -194,23 +194,24 @@ def create_groups_when_students_work_alone
   end
 
   def download_grouplist
-    assignment = Assignment.find(params[:assignment_id])
+    assignment = Assignment.where(id: params[:assignment_id]).first
 
     #get all the groups
     groupings = assignment.groupings #FIXME: optimize with eager loading
 
-    file_out = CSV.generate do |csv|
-       groupings.each do |grouping|
-         group_array = [grouping.group.group_name, grouping.group.repo_name]
-         # csv format is group_name, repo_name, user1_name, user2_name, ... etc
-         grouping.student_memberships.includes(:user).each do |member|
-            group_array.push(member.user.user_name)
-         end
-         csv << group_array
-       end
-     end
-
-    send_data(file_out, type: 'text/csv', disposition: 'inline')
+    file_out = MarkusCSV.generate(groupings) do |grouping|
+      group_array = [grouping.group.group_name, grouping.group.repo_name]
+      # csv format is group_name, repo_name, user1_name, user2_name, ... etc
+      grouping.student_memberships.includes(:user).each do |member|
```

Comment:
Please see my note in the other downloads PR. This may affect future PRs in the downloads.